### DISCUSSION TITLE
Is it possible to communicate send AMD GPU data directly from Windows machine to Linux

### DISCUSSION THREAD

**Discussion by sen-sd** [ofiwg/libfabric#8293] [Ideas]
My doubt is with libfabric can we send GPU data from Windows to Linux with out CPU copy.
In our environment we are using Mellanox ConnectX-5 for RDMA and GPU we are using Radeon WX7100.

I checked libfabric code and found that for RDMA for windows Libfabric internally using NetworkDirect. 
And it is using IND2MemoryRegion::Register() function to register application allocated memory and no option found to send GPU buffer. Please help

Client - Windows -> Windows 10, Mellanox ConnectX-5 , Radeon WX7100
Server - Linux -> Mellanox ConnectX-5

**Comment by shefty** [comment]
I'm not aware of Windows having OS support to setup the PCI peer to peer transfers that are needed for a NIC to access GPU memory directly.  The Linux feature for this is called dma-buf.  There's no Windows OS equivalent.

**Reply by sen-sd** [reply]
Thank you for reply. So hope in Windows , Nvidia GPU also will not support DMA.
In libfabric  source found CUDA support, Is this only for Linux.
